discovery-provider/src/challenges/challenge.py

```diff
@@ -71,6 +71,7 @@ def process(self, session, event_type, event_metadatas):
             return
 
         user_ids = list(map(lambda x: x["user_id"], event_metadatas))
+        user_id_metadatas = { x["user_id"]: x for x in event_metadatas}
```

Comment:
inconsistent curly brace spacing. are you not running linter on save??? this should get autoformatted :P